### PR TITLE
Use yarn and move Momentjs from gemfile to yarn

### LINKED ISSUE #5120: Use NPM or add YARN to manage JS packages
### What we should change and why (this is tech debt)
It's 2020 and it's a bit ridiculous we have our JS dependencies on our codebase :see_no_evil: 
Using the Gemfile (like we do for angularjs-file-upload-rails for example) isnt the right approach because we depend on out of date or nonexistent rails wrappers for the JS dependencies.
The correct and simple solution is to use a JS package manager and integrate it into the asset pipeline.

There are two options here that I think are valid, use npm.
You can run
`npm install --save textangular`
and that is probably it as this will be available to be used in app/assets/javascripts/admin/all.js

But yarn is better than npm for a few reasons so we could start using yarn for this and get the dependencies through it.

This can then be used to include all the JS dependencies in our repo:
- app/assets/javascripts/shared
- app/assets/stylesheets/shared
- vendor/assets/javascripts/
- vendor/assets/stylesheets/
- Gemfile

### Context
This came up in quite a few PRs where we are upgrading dependencies directly in our repo:
- upgrading text angular in Feb 2018: https://github.com/openfoodfoundation/openfoodnetwork/pull/2060#issuecomment-364200533
- upgrade text angular in Feb 2020 - https://github.com/openfoodfoundation/openfoodnetwork/pull/4816#issuecomment-592457713

### Impact and timeline
This will be a very good clean up of our frontend code.

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #5120

We have added npm and yarn to our toolset in openfoodfoundation/ofn-install#613.
All servers are now running yarn and on every deploy yarn install is executed to install dependencies.
No dependencies are currently installed with yarn (there are some **dev** dependencies in yarn but they are not deployed to staging or production servers).

This PR removes the Gemfile dependency momentjs and adds momentjs as a npm/yarn dependency :tada:

With yarn we get a js dependency equivalent to Gemfile.lock, yarn.lock:
image

From now on we dont need to add js dependencies to our repo or to Gemfile, we can just use "yarn add" and require them in rails app/assets using their path on node_modules.
See post in discourse to clarify how to do this:
https://community.openfoodnetwork.org/t/better-js-dependency-management-with-yarn/2057

Locally, developers will be able to keep using only npm except if they want to add a new JS dependency, they will need to install yarn locally.
I used this guide to install yarn locally and it just worked:
https://linuxize.com/post/how-to-install-yarn-on-ubuntu-18-04/

#### What should we test?
We should test momentjs and verify that time related translations are working, for example in the OC closing message in the shopfront:
image

In this PR, I also added the missing languages to moment: phi and nl, this last one will be seen in the BE instance where it is currently broken: https://openfoodnetwork.be/magasin-du-monde-oxfam-jodoigne-commerce-equitable/shop?locale=nl_BE#/shop

#### Release notes
Added yarn to our toolset to manage JS dependencies (instead of using gem wrappers or copying the code of all JS dependencies to our repo) and move momentjs from a gemfile dependency to a yarn dependency.
Changelog Category: Changed

### CODE REVIEW

**Comment by andresgutgon** [comment]
Should be this done when we migrate to new Rails that comes with Webpack as default javascript bundle?

**Comment by luisramos0** [comment]
Hi @andresgutgon  thanks for your question!
It's rails 5.1, it will be a while until then for us on 4.0.13 right now.

But I wonder if you understand how this is going to fit into webpacker, because I read:
"Rails 5.1 ships with webpack and yarn"
https://medium.com/statuscode/introducing-webpacker-7136d66cddfb

What I read here is: if yarn is there, we will find a way to use our package.json/yarn.lock when we get to rails 5.1 and webpacker. 
I am not totally sure but I think this is the way to go now.

**Comment by luisramos0** [comment]
I think it could be as easy as "webpack calls yarn install by default"
"You will also find webpacker gem in the Gemfile by default. The rails new command will also install tons of npm packages via yarn."
https://prathamesh.tech/2019/08/26/understanding-webpacker-in-rails-6/

**Comment by luisramos0** [comment]
some problem in the build, I need to have a look and make sure node_modules are seen when the specs are running.

File: package.json [hunk 1]
```diff
@@ -1,16 +1,19 @@
 {
   "name": "openfoodnetwork",
-  "version": "1.7.1",
+  "version": "1.0.0",
```
Comment (renanbona) [inline package.json:3]: Can I ask why the version was set to 1.0.0?

Comment (luisramos0) [reply package.json:3]: hey Renan, welcome to OFN!
the commit mesage explains:
https://github.com/openfoodfoundation/openfoodnetwork/pull/6049/commits/fe7d7b54d10258da884775ac12475e09572090ea

I havent tried to remove the version entry, maybe that would be better.

**Comment by luisramos0** [comment]
ok, buidl fixed, all I needed to do was to add a new commend to each semaphore build job: yran install :ok_hand: 
Ready for review.

**Review by sauloperez — APPROVED** [review]

**Comment by sauloperez** [comment]
What a nice trick!

**Comment by Matt-Yorkley** [comment]
Cool, I just pulled this branch and gave it a try. Initially, starting the rails server in dev gives this:

Screenshot from 2020-10-07 11-25-52

After installing yarn locally and running `yarn install`, everything is fine :+1:

I think we should add these steps to the wiki for setting up a dev environment:

```
# Install yarn
curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
sudo apt update && sudo apt install yarn

# Fetch dependencies with yarn
yarn install
```

**Review by Matt-Yorkley — APPROVED** [review]
Awesome :tada:

**Comment by filipefurtad0** [comment]
Hey @luisramos0 ,

I had a look at the how the OC closing message looked like before:

image

And after staging:

image

All good here.

Regarding phi and nl languages, did the following:
- logged into the server
- edited the application.yml file, adding `fil_PH` and `nl_BE` to the `AVAILABLE_LOCALES`
- restarted the unicorn service
- checked the /shop again:

image

This looks good now :tada: 

Ready to go!

**Comment by luisramos0** [comment]
It's awesome you know how to do this now when needed!

image

**Comment by filipefurtad0** [comment]
You once taught me how to log in!! :-D
I'm just standing in the shoulders of giants...

**Pull request merged**